pennylane/qnode_new/jacobian.py

```diff
@@ -0,0 +1,528 @@
+                successor_ops = self._op_descendants(op, 'G')
+                if not all(x.supports_heisenberg for x in successor_ops):
+                    non_gaussian_ops = [x for x in successor_ops if not x.supports_heisenberg]
+                    # a non-Gaussian successor is OK if it isn't succeeded by any observables
+                    for x in non_gaussian_ops:
+                        if self._op_descendants(x, 'E'):
+                            return 'F'
+
+                # check successor EVs, if any order-2 observables are found return 'A2', else return 'A'
+                for observable in obs_successors:
+                    if observable.ev_order is None:
+                        # ev_order of None corresponds to a non-Gaussian observable
+                        return 'F'
+                    if observable.ev_order == 2:
+                        if observable.return_type is ObservableReturnTypes.Variance:
+                            # second order observables don't support
+                            # analytic diff of variances
+                            return 'F'
+                        op.grad_method = 'A2'  # bit of a hack
+                return 'A'
+            return op.grad_method
+
+        # operations that depend on this free parameter
+        ops = [d.op for d in self.variable_deps[idx]]
+        methods = list(map(best_for_op, ops))
+
+        if all(k in ('A', 'A2') for k in methods):
+            return 'A'
+
+        if None in methods:
+            return None
+
+        return 'F'
+
+    def jacobian(self, args, kwargs=None, *, wrt=None, method='B', options=None):
+        r"""Compute the Jacobian of the QNode.
+
+        Returns the Jacobian of the parametrized quantum circuit encapsulated in the QNode.
+        The Jacobian is returned as a two-dimensional array. The (possibly nested) input arguments
+        of the QNode are :func:`flattened <_flatten>` so the QNode can be interpreted as a simple
+        :math:`\mathbb{R}^m \to \mathbb{R}^n` function.
+
+        The Jacobian can be computed using several methods:
+
+        * Finite differences (``'F'``). The first order method evaluates the circuit at
+          :math:`n+1` points of the parameter space, the second order method at :math:`2n` points,
+          where ``n = len(wrt)``.
+
+        * Analytic method (``'A'``). Works for all one-parameter gates where the generator
+          only has two unique eigenvalues; this includes one-parameter single-qubit gates.
+          Additionally, can be used in CV systems for Gaussian circuits containing only first-
+          and second-order observables.
+          The circuit is evaluated twice for each incidence of each parameter in the circuit.
+
+        * Best known method for each parameter (``'B'``): uses the analytic method if
+          possible, otherwise finite difference.
+
+        * Device method (``'D'``): Delegates the computation of the Jacobian to the
```

Comment:
Nice!

Reply:
Does not appear so (no code related to this method). But it's in the documentation there

Reply:
🤔

Reply:
Yes, I think we merged `CustomGradientQNode` with `JacobianQNode` in the ADR. The `device.jacobian` call is still missing, I'll add it ASAP.